packages/@rescript/runtime/Stdlib_Float.res

```diff
@@ -58,11 +58,11 @@ external toPrecisionWithPrecision: (float, ~digits: int) => string = "toPrecisio
 external toStringWithRadix: (float, ~radix: int) => string = "toString"
 @send external toLocaleString: float => string = "toLocaleString"
 
-let fromString = i =>
-  switch parseFloat(i) {
-  | i if isNaN(i) => None
-  | i => Some(i)
-  }
+let fromString: string => option<float> = %raw(`str => {
+  if (!str || !str.trim()) return;
+  let num = +str;
+  return isNaN(num) ? undefined : num;
+}`)
```

Comment:
I wrote some `fast-check` tests and couldn't find a difference between `+str` and `Number(str)`, but the latter is easier to understand and could be written in pure ReScript.

How about converting this function to ReScript?

```rescript
%%private(
  @val external makeNumber: 'a => float = "Number"
  @send external trim: string => string = "trim"
)
let fromString: string => option<float> = str =>
  if Js.typeof(str) === "string" && trim(str) !== "" {
    let num = makeNumber(str)
    isNaN(num) ? None : Some
  } else {
    None
  }
```